pkg/ha/ha_parser.go

```diff
@@ -86,6 +86,14 @@ func (h *haParser) ParseData(tts []prompb.TimeSeries) (map[string][]model.Sample
 			continue
 		}
 
+		// Drop __replica__ labelSet from samples
+		for ind, value := range t.Labels {
+			if value.Name == model.ReplicaNameLabel {
+				t.Labels = append(t.Labels[:ind], t.Labels[ind+1:]...)
```

Comment:
we have no guarantee that the replica name label will always be in the same position between samples right?

Reply:
In my observation, I haven't noticed it. That's the reason we are iterating over labels to find `cluster` & `replica` labels too for HA logic.